Sources/Purchasing/TrialOrIntroPriceEligibilityChecker.swift

```diff
@@ -267,3 +287,68 @@ extension TrialOrIntroPriceEligibilityChecker {
 // @unchecked because:
 // - Class is not `final` (it's mocked). This implicitly makes subclasses `Sendable` even if they're not thread-safe.
 extension TrialOrIntroPriceEligibilityChecker: @unchecked Sendable {}
+
+// MARK: - Diagnostics
+
+private extension TrialOrIntroPriceEligibilityChecker {
+
+    func trackTrialOrIntroEligibilityRequestIfNeeded(startTime: Date,
+                                                     requestedProductIds: Set<String>,
+                                                     result: [String: IntroEligibility],
+                                                     error: Error?,
+                                                     storeKitVersion: StoreKitVersion) {
+        guard #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *),
+              let diagnosticsTracker = self.diagnosticsTracker else {
+            return
+        }
+
+        var unknownCount, ineligibleCount, eligibleCount, noIntroOfferCount: Int?
+        if !result.isEmpty {
+            (unknownCount, ineligibleCount, eligibleCount, noIntroOfferCount) = result.reduce(into: (0, 0, 0, 0)) {
+                switch $1.value.status {
+                case .unknown:
+                    $0.0 += 1
+                case .ineligible:
+                    $0.1 += 1
+                case .eligible:
+                    $0.2 += 1
+                case .noIntroOfferExists:
+                    $0.3 += 1
+                }
+            }
+        }
+
+        let errorCode: Int?
+        let errorMessage: String?
+        switch error {
+        case let purchasesError as PurchasesError:
+            errorCode = purchasesError.errorCode
+            errorMessage = purchasesError.localizedDescription
+        case let purchasesErrorConvertible as PurchasesErrorConvertible:
+            let purchasesError = purchasesErrorConvertible.asPurchasesError
+            errorCode = purchasesError.errorCode
+            errorMessage = purchasesError.localizedDescription
+        case let receiptParserError as PurchasesReceiptParser.Error:
+            errorCode = ErrorCode.invalidReceiptError.rawValue
+            errorMessage = receiptParserError.errorDescription ?? receiptParserError.localizedDescription
+        case let otherError:
+            errorCode = otherError != nil ? ErrorCode.unknownError.rawValue : nil
+            errorMessage = otherError?.localizedDescription
+        }
```

Comment:
I'm not really sure about this implementation for `errorCode` and `errorMessage`, but based on the errors thrown in the implementations, I think that this could include useful information about the errors

Reply:
That's a very good point. I think we could do what you propose, to have a unified way of sending `errorMessage` and `errorCode` parameters in all diagnostics events.

I'll try to solve this thing in a separate PR. I think that, ideally, the same errors in different diagnostics events should be sent the same way.